sw/device/tests/rv_core_ibex_nmi_irq_test.c

```diff
@@ -0,0 +1,263 @@
+#include "sw/device/lib/testing/rand_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_isrs.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+typedef void (*isr_handler)(void);
+
+static volatile isr_handler expected_isr_handler;
+static volatile dif_rv_core_ibex_nmi_state_t nmi_state;
+static volatile bool nmi_fired = false;
+
+static dif_pwrmgr_t pwrmgr;
+static dif_rv_core_ibex_t rv_core_ibex;
+static dif_aon_timer_t aon_timer;
+static aon_timer_isr_ctx_t aon_timer_ctx = {
+    .aon_timer = &aon_timer,
+    .plic_aon_timer_start_irq_id =
+        kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired,
+};
+
+static dif_rv_plic_t plic;
+static plic_isr_ctx_t plic_ctx = {
+    .rv_plic = &plic,
+    .hart_id = kTopEarlgreyPlicTargetIbex0,
+};
+
+static volatile dif_aon_timer_irq_t irq;
+static dif_alert_handler_t alert_handler;
+static alert_handler_isr_ctx_t alert_handler_ctx = {
+    .alert_handler = &alert_handler,
+    .plic_alert_handler_start_irq_id = kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+    .expected_irq = kDifAlertHandlerIrqClassb,
+    .is_only_irq = false,
+};
+
+/**
+ * Program the alert handler to escalate on alerts upto phase 2 (i.e. reset)
+ * but at the phase 1 (i.e. wipe secrets)the nmi interrupt handler should clear
+ * the escalation.
+ */
+enum {
+  kEscalationPhase0Micros = 100 * 1000,  // 10 ms
+  kEscalationPhase2Micros = 100,         // 100 us
+};
+
+static void alert_handler_config(void) {
+  dif_alert_handler_alert_t alerts[] = {kTopEarlgreyAlertIdPwrmgrAonFatalFault};
+  dif_alert_handler_class_t alert_classes[] = {kDifAlertHandlerClassA};
+
+  dif_alert_handler_escalation_phase_t esc_phases[] = {
+      {.phase = kDifAlertHandlerClassStatePhase0,
+       .signal = 0,
+       .duration_cycles = udiv64_slow(
+           kEscalationPhase0Micros * kClockFreqPeripheralHz, 1000000, NULL)},
+      {.phase = kDifAlertHandlerClassStatePhase2,
+       .signal = 3,
+       .duration_cycles = udiv64_slow(
+           kEscalationPhase2Micros * kClockFreqPeripheralHz, 1000000, NULL)}};
+
+  dif_alert_handler_class_config_t class_config[] = {{
+      .auto_lock_accumulation_counter = kDifToggleDisabled,
+      .accumulator_threshold = 0,
+      .irq_deadline_cycles =
+          udiv64_slow(10 * kClockFreqPeripheralHz, 1000000, NULL),
+      .escalation_phases = esc_phases,
+      .escalation_phases_len = ARRAYSIZE(esc_phases),
+      .crashdump_escalation_phase = kDifAlertHandlerClassStatePhase3,
+  }};
+
+  dif_alert_handler_class_t classes[] = {kDifAlertHandlerClassA};
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
```

Comment:
s/`alert_classes`/`{kDifAlertHandlerClassA}` Or move the definition of `alert_classes` closer to this definition to improve readability.

Reply:
I moved the definition closer as the struct expects a pointer.